test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestApiParser.java

```diff
@@ -146,4 +145,17 @@ public ClientYamlSuiteRestApi parse(String location, XContentParser parser) thro
 
         return restApi;
     }
+    
+    private boolean getRequired(XContentParser parser) throws IOException {
+        boolean required = false;
+        while(parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            if (parser.currentToken() == XContentParser.Token.FIELD_NAME) {
+                if ("required".equals(parser.currentName())) {
+                    parser.nextToken();
+                    required = parser.booleanValue();
+                }
```

Comment:
Can you either replace this with `CosntructingObjectParser<Boolean>` or manually throw `IOException`s for things other than `required` and for tokens other than `FIELD_NAME`?

Reply:
@nik9000 thanks for the hint! (I did not know of this parser)
I think that `CosntructingObjectParser<Boolean>` cannot be used here as the `required` field is an optional one.

I couldn't find an example how to use `ConstructingObjectParser` for the primitive wrapper classes. Do you have one?